cmd/gokr-packer/certs.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/gokrazy/internal/config"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func generateAndSignCert() ([]byte, *rsa.PrivateKey, error) {
+	notBefore := time.Now()
+	notAfter := notBefore.Add(2 * 365 * 24 * time.Hour)
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, err
+	}
+	template := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Organization: []string{"gokrazy"},
+		},
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+		DNSNames:              []string{*hostname},
+	}
+	priv, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, nil, err
+	}
+	pub := &priv.PublicKey
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, pub, priv)
+	if err != nil {
+		return nil, nil, err
+	}
+	return derBytes, priv, err
+}
+
+func getCertificate() (string, string, error) {
+	hostConfigPath := config.HostnameSpecific(*hostname)
+	var certPath, keyPath string
+	switch *useTLS {
+	case "self-signed":
+		certPath = filepath.Join(string(hostConfigPath), "cert.pem")
+		keyPath = filepath.Join(string(hostConfigPath), "key.pem")
+		gen := false
+		exist := true
+		if _, err := os.Stat(certPath); os.IsNotExist(err) {
+			gen = true
+			exist = false
+		}
+		if _, err := os.Stat(keyPath); os.IsNotExist(err) {
+			gen = true
+			exist = false
+		}
+		if exist {
+			// TODO: Check validity dates of existing certificate
+		}
+		if gen {
```

Comment:
Can you refactor this branch into a separate helper function please, to get rid of the 2-level indention here?